Sources/ArcGISToolkit/Components/Compass/Compass.swift

```diff
@@ -63,65 +55,49 @@ public struct Compass: View {
                 .aspectRatio(1, contentMode: .fit)
                 .opacity(opacity)
                 .frame(width: size, height: size)
-                .onAppear { opacity = shouldHide ? 0 : 1 }
-                .onChange(of: heading) { _ in
-                    let newOpacity: Double = shouldHide ? .zero : 1
+                .onAppear { opacity = shouldHide(forHeading: heading) ? 0 : 1 }
+                .onChange(of: heading) { newHeading in
+                    let newOpacity: Double = shouldHide(forHeading: newHeading) ? .zero : 1
                     guard opacity != newOpacity else { return }
-                    withAnimation(.default.delay(shouldHide ? 0.25 : 0)) {
+                    withAnimation(.default.delay(shouldHide(forHeading: newHeading) ? 0.25 : 0)) {
                         opacity = newOpacity
                     }
                 }
                 .onTapGesture {
-                    if let mapViewProxy {
-                        Task { await mapViewProxy.setViewpointRotation(0) }
-                    } else {
-                        heading = .zero
-                    }
+                    Task { await mapViewProxy?.setViewpointRotation(0) }
                 }
                 .accessibilityLabel("Compass, heading \(Int(heading.rounded())) degrees \(CompassDirection(heading).rawValue)")
         }
     }
 }
 
-public extension Compass {
-    /// Creates a compass with a binding to a viewpoint rotation (0° indicates
-    /// a direction toward true North, 90° indicates a direction toward true
-    /// West, etc.).
-    /// - Parameters:
-    ///   - viewpointRotation: The viewpoint rotation whose value determines the
-    ///   heading of the compass.
-    ///   - mapViewProxy: The proxy to provide access to map view operations.
-    init(
-        viewpointRotation: Binding<Double>,
-        mapViewProxy: MapViewProxy? = nil
-    ) {
-        let heading = Binding(get: {
-            viewpointRotation.wrappedValue.isZero ? .zero : 360 - viewpointRotation.wrappedValue
-        }, set: { newHeading in
-            viewpointRotation.wrappedValue = newHeading.isZero ? .zero : 360 - newHeading
-        })
-        self.init(heading: heading, mapViewProxy: mapViewProxy)
+extension Compass {
+    /// Returns a Boolean value indicating whether the compass should hide based on the
+    /// provided heading and whether the compass has been configured to automatically hide.
+    /// - Parameter heading: The heading used to determine if the compass should hide.
+    /// - Returns: `true` if the compass should hide, `false` otherwise.
+    func shouldHide(forHeading heading: Double) -> Bool {
+        (heading.isZero || heading.isNaN) && autoHide
     }
-    
-    /// Creates a compass with a binding to an optional viewpoint.
+}
+
+public extension Compass {
+    /// Creates a compass with a rotation (0° indicates a direction toward true North, 90° indicates
+    /// a direction toward true West, etc.).
```

Comment:
Is it me or is our doc confusing on this? Yes at a rotation of 90° the needle points west but the map is oriented to east.

Reply:
The doc talks specifically about the direction of rotation, which is correct in this regard (positive values counter-clockwise).